utils/misc.ml

```diff
@@ -17,6 +17,13 @@ exception Fatal_error
 let fatal_error msg =
   prerr_string ">> Fatal error: "; prerr_endline msg; raise Fatal_error
 
+let fatal_errorf fmt =
+  let (_ : string) = Format.flush_str_formatter () in
+  Format.kfprintf (fun ppf ->
+      fatal_error (Format.flush_str_formatter ()))
+    Format.str_formatter
```

Comment:
Flushing the string formatter and ignoring the result means that there is a risk of data loss. You should use `Format.formatter_of_buffer (Buffer.create 100)` instead.